CodePurpose/political_leaning_api
# User login route

## Details of Changes
User can now log in and log out. Auth has also been implemented.

- https://trello.com/c/e54slxP0/33-create-login-endpoint
- Add Description of Changes
  

## Testing
- run iex -S mix phx.server
- insert a user into DB either manually or simply run `mix run priv/repo/seeds.exs` or insert in iex ie:
```
changeset = User.changeset(%User{}, %{username: "someuser", email: "email@email.com", password: "pass", name: "name"}
Repo.insert(changeset)
```
- POST to localhost:4000/sessions/sign_in with json body:
```
{
  "email":"insert email here",
  "password": "insert passoword here"
}
```
- verify you receive a token
- copy token and make a GET request with the Authorization "Bearer {insert token here}" to the temporary endpoint: localhost:4000/secret/super to verify you have access
- make a DELETE request to `localhost:4000/sessions/sign_out` with the Auth header like in the above request
- verify auth token has been deleted by making a request to the super secret endpoint and you should be denied access
## Due Diligence

- [x ] I have added the details of changes.
- [ x] My code follows the code style of this project.
- [ x] My change requires a change to the documentation.
- [ x] I have added tests to cover my changes.
